json11.cpp

```diff
@@ -364,13 +364,52 @@ struct JsonParser {
             i++;
     }
 
+    /* consume_comment()
+     *
+     * Advance comments (c-style inline and multiline).
+     */
+    void consume_comment() {
+      if (str[i] == '/') {
+        i++;
+        if (str[i] == '/') { // inline comment
+          i++;
+          // advance until next line
+          while (str[i] != '\n')
+            i++;
+          consume_whitespace();
+          consume_comment();
+        }
+        else if (str[i] == '*') { // multiline comment
+          i++;
+           // advance until closing tokens
+          while (!(str[i] == '*' && str[i+1] == '/'))
+            i++;
+          i += 2;
+          consume_whitespace();
+          consume_comment();
+        }
+      }
+    }
+
+    /* consume_garbage()
+     *
+     * Advance until the current character is non-whitespace and non-comment.
+     */
+    void consume_garbage() {
```

Comment:
I feel like a loop would be a safer approach than recursion for consuming more comments/whitespace, since that would avoid the risk that your stack frame grows (and could overflow) when parsing many comments.  I feel like this function is the right place to put a loop.  If consume_whitespace() and consume_comment() returned a bool to indicate whether they consumed anything (or you just looked at the value of i) it would be easy to decide when to end the loop.